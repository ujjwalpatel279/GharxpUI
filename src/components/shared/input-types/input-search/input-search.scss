// ========================================================================
// COMPONENTS/SHARED/INPUT-TYPES/CURRENCY
// ========================================================================

@import '~sassHelpers';

.c-input-search {
  position: relative;
  display: inline-block;

  svg {
    position: absolute;
    top: 50%;
    left: spacer(2);
    transform: translateY(-50%);
  }

  input {
    padding-left: spacer(8);
    box-shadow: inset 0px 3px 3px rgba(black, 0.1);
  }
}
