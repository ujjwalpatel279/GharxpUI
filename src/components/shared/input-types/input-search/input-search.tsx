import React, { FunctionComponent, KeyboardEvent } from 'react';
import { useForm } from 'react-hook-form';

import { Input } from '../';
import { Icon } from '../..';

import './input-search.scss';

interface InputSearchProps {
  id: string;
  onKeyUp?: (e: KeyboardEvent<HTMLInputElement>) => void;
  setValue?: ReturnType<typeof useForm>['setValue'];
  value?: string;
}

export const InputSearch: FunctionComponent<InputSearchProps> = ({ id, onKeyUp, setValue, value }) => (
  <div className="c-input-search">
    <Icon name="search" />
    <Input type="text" id={id} placeHolder="Search" onKeyUp={onKeyUp} setValue={setValue} value={value} />
  </div>
);
