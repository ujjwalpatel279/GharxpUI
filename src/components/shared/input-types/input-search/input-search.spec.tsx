import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';

import { InputSearch } from './input-search';

describe('Input Search Component', () => {
  afterEach(cleanup);

  it('renders the search component', () => {
    const { container } = render(<InputSearch id="amount" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('input value is initially empty and has a value once a value is entered', () => {
    const { queryByPlaceholderText } = render(<InputSearch id="search" />);

    const input = queryByPlaceholderText('Search') as HTMLInputElement;
    expect(input.value).toBe('');

    fireEvent.change(input, { target: { value: 'Search' } });
    expect(input.value).toBe('Search');
  });
});
