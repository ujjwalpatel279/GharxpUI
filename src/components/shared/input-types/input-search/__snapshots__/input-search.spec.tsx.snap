// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Search Component renders the search component 1`] = `
<div
  class="c-input-search"
>
  <svg
    class="c-icon"
    height="16"
    viewBox="0 0 16 16"
    width="16"
  >
    <path
      d="M14.722 13.031l-3.395-2.888c-0.351-0.316-0.726-0.461-1.030-0.447 0.801-0.939 1.286-2.157 1.286-3.488 0-2.969-2.406-5.375-5.375-5.375s-5.375 2.406-5.375 5.375 2.406 5.375 5.375 5.375c1.331 0 2.549-0.484 3.488-1.286-0.014 0.303 0.131 0.679 0.447 1.030l2.888 3.395c0.494 0.549 1.302 0.596 1.795 0.103s0.446-1.3-0.103-1.795zM6.208 9.792c-1.979 0-3.583-1.604-3.583-3.583s1.604-3.583 3.583-3.583 3.583 1.604 3.583 3.583-1.604 3.583-3.583 3.583z"
    />
  </svg>
  <input
    aria-invalid="false"
    class="c-input"
    id="amount"
    name="amount"
    placeholder="Search"
    type="text"
  />
</div>
`;
